src/parser/github-comment-module.ts

```diff
@@ -0,0 +1,216 @@
+          )
+        );
+      }
+      if (sorted.reviews.length) {
+        content.push(
+          generateContributionRow(
+            "Review",
+            "Comment",
+            sorted.reviews.length,
+            sorted.reviews.reduce((acc, curr) => acc.add(curr.score?.reward ?? 0), new Decimal(0))
+          )
+        );
+      }
+      return content.join("");
+    }
+
+    function createIncentiveRows() {
+      const content: string[] = [];
+
+      if (!sorted) {
+        return content.join("");
+      }
+
+      function buildIncentiveRow(commentScore: GithubCommentScore) {
+        // Properly escape carriage returns for HTML rendering
+        const formatting = stringify(commentScore.score?.formatting?.content).replace(/[\n\r]/g, "&#13;");
+        return `
+          <tr>
+            <td>
+              <h6>
+                <a href="${commentScore.url}" target="_blank" rel="noopener">${commentScore.content.replace(/(.{64})..+/, "$1…")}</a>
+              </h6>
+            </td>
+            <td>
+            <details>
+              <summary>
+                ${Object.values(commentScore.score?.formatting?.content || {}).reduce((acc, curr) => {
+                  return acc.add(curr.score * curr.count);
+                }, new Decimal(0))}
+              </summary>
+              <pre>${formatting}</pre>
+             </details>
+            </td>
+            <td>${commentScore.score?.relevance || "-"}</td>
+            <td>${commentScore.score?.reward || "-"}</td>
+          </tr>`;
+      }
+
+      for (const issueComment of sorted.issues.comments) {
+        content.push(buildIncentiveRow(issueComment));
+      }
+      for (const reviewComment of sorted.reviews) {
+        content.push(buildIncentiveRow(reviewComment));
+      }
+      return content.join("");
+    }
+
+    return `
+    <details>
+      <summary>
+        <b>
+          <h3>
+            <a href="${result.permitUrl}" target="_blank" rel="noopener">
+              [ ${result.total} ${getPayoutConfigByNetworkId(program.opts().evmNetworkId).symbol} ]
+            </a>
+          </h3>
+          <h6>
+            @${username}
+          </h6>
+        </b>
+      </summary>
+      <h6>Contributions Overview</h6>
+      <table>
+        <thead>
+          <tr>
+            <th>View</th>
+            <th>Contribution</th>
+            <th>Count</th>
+            <th>Reward</th>
+          </tr>
+        </thead>
+        <tbody>
+          ${createContributionRows()}
+        </tbody>
+      </table>
+      <h6>Conversation Incentives</h6>
+      <table>
+        <thead>
+          <tr>
+            <th>Comment</th>
+            <th>Formatting</th>
+            <th>Relevance</th>
+            <th>Reward</th>
+          </tr>
+        </thead>
+        <tbody>
+          ${createIncentiveRows()}
+        </tbody>
+      </table>
+    </details>
+    `
+      .replace(/\s+/g, " ")
```

Comment:
This regex appears to convert repeating spaces i.e. `"   "` to a single space `" "` what is the purpose of this?

Reply:
@gentlementlegen

Reply:
This matches any whitespace character (spaces, tabs, line breaks) so basically I make the produced HTML one liner. This helps to collect it from the CLI, makes the string shorter and more compact.